components/uniset/src/.rustfmt.toml

```diff
@@ -0,0 +1,5 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+max_width = 200                    # length of line
```

Comment:
`option`: is there a way to annotate a single file like this? I know that for generated files I can set `cargo clippy` lints, but I'm wondering if there's a better way than `rustfmt.toml` to tell `rust fmt` to back off.

Reply:
Good question. I checked again, and it looks like rustfmt tries to steer away from such granular configuration because it ["tries to work on as much Rust code as possible"](https://github.com/rust-lang/rustfmt#limitations). In spite of that, I tried applying the attributes mentioned in the [Tips section](https://github.com/rust-lang/rustfmt#tips), but the behavior allowed is very limited and did not come close to what we are thinking of. So currently, the answer is still no.

Reply:
I saw this in @gregtatum's PR#480: `#[rustfmt::skip]`

I found documentation for it: https://github.com/rust-lang/rustfmt#tips

It looks like you might be able to disable a certain rule for a certain file like this?

```rust
#[rustfmt::skip::macros(html)]
```

And I wonder if changing `#[` to `#![` would make it work for a whole file.

Reply:
I had already tried the `#![` syntax at the file level, and that didn't work. rustfmt only exposes `rustfmt::skip::macros` and `rustfmt::skip::attributes`. Hmm, I see now that there is skip formatting altogether with `rustfmt::skip`. I don't see anything like clippy where rustfmt would allow us to have the granularity of control to turn off a specific type of check (ex: `max_width`).

Reply:
Hmm.  It might be nice if we put whatever code that doesn't want to be formatted in its own files (with as little other code in those files as possible), and then just skipped the whole file.

Could we also annotate every line or every function with `rustfmt::skip`?

Let's discuss tomorrow.

Reply:
Do we want to still follow up on the rustfmt question?